Estib/logs archives with role delegation (#4862)
DataDog/documentation: content/en/logs/archives/s3.md

 
## Create and Configure an S3 Bucket

Go into your [AWS Console][1] and [create an S3 bucket][2] to send your archives to. Be careful not to make your bucket publicly readable. 

> Next, grant Datadog permissions to write log archives to your S3 bucket. You can grant those permissions either from your bucket policy or via role delegation. 

Reviewer: ```suggestion
Next, grant Datadog permissions to write log archives to your S3 bucket. You can grant those permissions either from your bucket policy or with role delegation. 
```
Avoiding the word via for translation purposes

    {{< img src="logs/archives/log_archives_s3_iam_policy.png" alt="IAM Policy for S3 Archives" responsive="true" style="width:75%;">}}

> 2. Go to your [Pipelines page in your Datadog application][1], and select the **Add archive** option at the bottom. Only Datadog users with admin status can complete this and the following step.

Reviewer: ```suggestion
2. Go to your [Pipelines page in Datadog][1], and select the **Add archive** option at the bottom. Only Datadog users with admin status can complete this and the following step.
```

3. Input your bucket name. Optionally input a prefix directory for all the content of your log archives. Save your archive, and you are finished. 

    {{< img src="logs/archives/log_archives_s3_datadog_settings.png" alt="Set your S3 bucket info in Datadog" responsive="true" style="width:75%;">}}

[1]: https://app.datadoghq.com/logs/pipelines
{{% /tab %}}

{{% tab "Role Delegation" %}}

> **Use of role delegation for writing log archives is in private beta. Ask support@datadoghq.com if you would like to have it enabled for your account.**

Reviewer: ```suggestion
**Use of role delegation for writing log archives is in private beta. Contact [Datadog support][4] to have it enabled for your account.**
```

1. Set up the [AWS integration][1] for the AWS account that holds your S3 bucket. This involves [creating a role][2] that Datadog can use to integrate with AWS Cloudwatch. 

2. Add the following permission statement to the IAM policies of your Datadog role (be sure to edit the bucket names and, if desired, specify the paths that contain your log archives):

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "DatadogUploadLogArchives",
            "Effect": "Allow",
            "Action": [
                "s3:PutObject"
            ],
            "Resource": [
                "arn:aws:s3:::<MY_BUCKET_NAME_1_/_MY_OPTIONAL_BUCKET_PATH_1>/*",
                "arn:aws:s3:::<MY_BUCKET_NAME_2_/_MY_OPTIONAL_BUCKET_PATH_2>/*"
            ]
        }
    ]
}
```

> 3. Go to your [Pipelines page in your Datadog application][3], and select the **Add archive** option at the bottom. Only Datadog users with admin status can complete this and the following step.

Reviewer: ```suggestion
3. Go to your [Pipelines page in Datadog][3], and select the **Add archive** option at the bottom. Only Datadog users with admin status can complete this and the following step.
```

> 4. Select the appopriate AWS Account + Role combination for your S3 bucket. Input your bucket name. Optionally input a prefix directory for all the content of your log archives. Save your archive, and you are finished. 

Reviewer: ```suggestion
4. Select the appopriate AWS account + role combination for your S3 bucket. Input your bucket name. Optionally input a prefix directory for all the content of your log archives. Save your archive, and you are finished. 
```

    {{< img src="logs/archives/log_archives_s3_datadog_settings_role_delegation.png" alt="Set your S3 bucket info in Datadog" responsive="true" style="width:75%;">}}
    
[1]: https://app.datadoghq.com/account/settings#integrations/amazon-web-services
[2]: /integrations/amazon_web_services/?tab=allpermissions#installation

> [3]: https://app.datadoghq.com/logs/pipelines

Reviewer: ```suggestion
[3]: https://app.datadoghq.com/logs/pipelines
[4]: /help
```
